src/HDF5.jl

```diff
@@ -222,6 +222,16 @@ const H5T_NATIVE_DOUBLE   = read_const(:H5T_NATIVE_DOUBLE_g)
 # Library versions
 const H5F_LIBVER_EARLIEST = 0
 const H5F_LIBVER_LATEST   = 1
+# Constructed types (occurs at runtime)
+function make_float16()
+    FLOAT16 = h5t_copy(H5T_NATIVE_FLOAT)
+    h5t_set_fields(FLOAT16, 15, 10, 5, 0, 10)
+    h5t_set_size(FLOAT16, 2)
+    h5t_set_ebias(FLOAT16, 15)
+    h5t_lock(FLOAT16)
+    return FLOAT16
+end
+# const H5T_FLOAT16 = make_float16()  (in `__init__()`)
```

Comment:
I guess I don't really understand why this has to be in `__init__` do you think you could please explain this to me?

Reply:
This seems quite possible. Another library (e.g. PETSc) could depend on HDF5_jll so that HDF5_jll is loaded and initialized before HDF5.jl is initialized. It would be quite difficult to trigger this in a test case though.